pkg/chunk/cache/diskcache.go

```diff
@@ -33,14 +72,25 @@ func (cfg *DiskcacheConfig) RegisterFlags(f *flag.FlagSet) {
 
 // Diskcache is an on-disk chunk cache.
 type Diskcache struct {
-	mtx     sync.RWMutex
 	f       *os.File
 	buckets uint32
 	buf     []byte
+	index   *diskcacheIndex
 }
 
 // NewDiskcache creates a new on-disk cache.
 func NewDiskcache(cfg DiskcacheConfig) (*Diskcache, error) {
+	if globalCache == nil {
+		var err error
+		globalCache, err = newDiskcache(cfg)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return globalCache, nil
```

Comment:
I don't think this is neccesarily safe - can we guarantee there isn't any concurrency here?  I'd suggest you do it in a sync.Once.

Reply:
Will do